src/styles/popover.scss

```diff
@@ -279,82 +293,151 @@ $block: #{$fd-namespace}-popover;
+      &--top {
+        &::after {
+          margin: 0 (-$fd-popover-arrow-width-half);
+        }
+      }
 
-    &[x-placement^="top"] {
-      margin-bottom: $fd-popover-arrow-height;
+      &--start,
+      &--end {
+        &::after {
+          margin: (-$fd-popover-arrow-width-half) 0;
+        }
+      }
 
-      .#{$block}__arrow {
+      &--bottom {
         bottom: -$fd-popover-arrow-height;
-        margin: 0 $fd-popover-arrow-width-half;
         border-width: $fd-popover-arrow-height $fd-popover-arrow-width-half 0 $fd-popover-arrow-width-half;
         border-top-color: $fd-popover-arrow-color;
 
         &::after {
           bottom: 1px;
-          margin: 0 (-$fd-popover-arrow-width-half);
           border-width: $fd-popover-arrow-height $fd-popover-arrow-width-half 0 $fd-popover-arrow-width-half;
           border-top-color: $fd-popover-background-color;
         }
       }
-    }
 
-    &[x-placement^="bottom"] {
-      margin-top: $fd-popover-arrow-height;
-
-      .#{$block}__arrow {
+      &--top {
         top: -$fd-popover-arrow-height;
-        margin: 0 $fd-popover-arrow-width-half;
         border-width: 0 $fd-popover-arrow-width-half $fd-popover-arrow-height $fd-popover-arrow-width-half;
         border-bottom-color: $fd-popover-arrow-color;
 
         &::after {
           top: 1px;
-          margin: 0 (-$fd-popover-arrow-width-half);
           border-width: 0 $fd-popover-arrow-width-half $fd-popover-arrow-height $fd-popover-arrow-width-half;
           border-bottom-color: $fd-popover-background-color;
         }
       }
-    }
 
-    &[x-placement^="left"] {
-      margin-right: $fd-popover-arrow-height;
-
-      .#{$block}__arrow {
+      &--end {
         right: -$fd-popover-arrow-height;
-        margin: $fd-popover-arrow-width-half 0;
         border-width: $fd-popover-arrow-width-half 0 $fd-popover-arrow-width-half $fd-popover-arrow-height;
         border-left-color: $fd-popover-arrow-color;
 
         &::after {
           right: 1px;
-          margin: -$fd-popover-arrow-width-half 0;
           border-width: $fd-popover-arrow-width-half 0 $fd-popover-arrow-width-half $fd-popover-arrow-height;
           border-left-color: $fd-popover-background-color;
         }
       }
-    }
-
-    &[x-placement^="right"] {
-      margin-left: $fd-popover-arrow-height;
 
-      .#{$block}__arrow {
+      &--start {
         left: -$fd-popover-arrow-height;
-        margin: $fd-popover-arrow-width-half 0;
         border-width: $fd-popover-arrow-width-half $fd-popover-arrow-height $fd-popover-arrow-width-half 0;
         border-right-color: $fd-popover-arrow-color;
 
         &::after {
           left: 1px;
-          margin: -$fd-popover-arrow-width-half 0;
           border-width: $fd-popover-arrow-width-half $fd-popover-arrow-height $fd-popover-arrow-width-half 0;
           border-right-color: $fd-popover-background-color;
         }
       }
+
+      &-x--start {
+        left: $fd-popover-arrow-offset;
```

Comment:
shouldn't this have RTL?

Reply:
No, it shouldn't. See the example.

<img width="468" alt="image" src="https://user-images.githubusercontent.com/20265336/157039154-0e652628-c7bb-49cf-a776-602734aa4c77.png">

Reply:
Appropriate css classes will be generated when initializing popover, then `start` will be `end` in RTL and vice versa.